monai/engines/workflow.py

```diff
@@ -162,7 +162,20 @@ def _register_post_transforms(self, posttrans: Callable):
 
         @self.on(IterationEvents.MODEL_COMPLETED)
         def run_post_transform(engine: Engine) -> None:
-            engine.state.output = apply_transform(posttrans, engine.state.output)
+            if isinstance(engine.state.batch, dict) and isinstance(engine.state.output, dict):
```

Comment:
this could be a utility as we discussed

Reply:
Sure, updated in the latest commit.
Thanks.